modules/plugins/jboss-as-7/src/main/java/org/rhq/modules/plugins/jbossas7/helper/JBossCliConfiguration.java

```diff
@@ -0,0 +1,243 @@
+
+/**
+ * A JBoss CLI configuration - loaded from jboss-cli.xml
+ *
+ * @author Libor Zoubek
+ */
+public class JBossCliConfiguration {
+
+    private final Log log = LogFactory.getLog(JBossCliConfiguration.class);
+
+    private Document document;
+    private XPathFactory xpathFactory;
+    private final File jbossCliXml;
+    private final ServerPluginConfiguration serverConfig;
+    /**
+     *
+     * @param jbossCliXml absolute path to jboss-cli.xml file
+     */
+    public JBossCliConfiguration(File jbossCliXml, ServerPluginConfiguration serverConfig) throws Exception {
+        this.jbossCliXml = jbossCliXml;
+        this.serverConfig = serverConfig;
+        DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
+        DocumentBuilder builder = factory.newDocumentBuilder();
+        InputStream is = new FileInputStream(jbossCliXml);
+        try {
+            this.document = builder.parse(is);
+        } finally {
+            is.close();
+        }
+        this.xpathFactory = XPathFactory.newInstance();
+    }
+
+    private Comment createComment() {
+        return this.document.createComment(" added by RHQ plugin ");
+    }
+
+    /**
+     * Setup SSL configuration properties by reading it from HostConfiguration (XML File) and expecting VAULT 
+     * to be present. 
+     * @param hostConfig
+     * @return
+     */
+    public String configureSecurityUsingVault(HostConfiguration hostConfig) {
+        Map<String, String> vaultOptions = hostConfig.getVault();
+        if (vaultOptions == null) {
+            return "Vault definition was not found in server configuration file";
+        }
+        TruststoreConfig serverIdentity = hostConfig.getServerIdentityKeystore();
+        if (serverIdentity == null) {
+            return "Could not find ssl configuration for management interface";
+        }
+        Node sslNode = (Node) xpathExpression("/jboss-cli/ssl", XPathConstants.NODE);
+        // clean-up existing ssl node
+        if (sslNode != null) {
+            this.document.getDocumentElement().removeChild(sslNode);
+        }
+        sslNode = addChildElement(document.getDocumentElement(), "ssl");
+        sslNode.appendChild(createComment());
+
+        Node vaultNode = this.document.createElement("vault");
+        sslNode.appendChild(vaultNode);
+
+        for (Entry<String, String> vaultOpt : vaultOptions.entrySet()) {
+            Element opt = addChildElement(vaultNode, "vault-option");
+            opt.setAttribute("name", vaultOpt.getKey());
+            opt.setAttribute("value", vaultOpt.getValue());
+        }
+
+        addChildElement(sslNode, "trust-store", serverIdentity.alias);
+        addChildElement(sslNode, "trust-store", serverIdentity.path);
+        // in standalone.xml vault value is referred as ${VAULT:...} we need to strip it for jboss-cli.xml
+        addChildElement(sslNode, "trust-store-password", stripBrackets(serverIdentity.keystorePassword));
```

Comment:
IMHO it is. According to [docs](https://developer.jboss.org/wiki/JBossAS7SecuringPasswords), vaults have to be enclosed in ${} in standalone.xml file, and [here](https://access.redhat.com/documentation/en-US/JBoss_Enterprise_Application_Platform/6.3/html-single/Security_Guide/index.html) must not be in jboss-cli.xml file. 

My testing also confirms this. 

I would ignore a potential case when someone can store password into system property, refer to it using ${} and then run this new operation to setup jboss-cli.xml using vault.